modules/apps/portal/portal-util-test/src/testIntegration/java/com/liferay/portal/util/test/PortalImplAlternateURLTest.java

```diff
@@ -254,6 +254,13 @@ public void testAlternateURLWithLayout() throws Exception {
 			availableLocales, defaultLocale, friendlyURLMap, 3, themeDisplay);
 	}
 
+	@Test
+	public void testAlternateURLWithRuntimeExceptionURL() throws Exception {
+		_testAlternateURLWithRuntimeExceptionURL("/guest/p");
+
+		_testAlternateURLWithRuntimeExceptionURL("/guest/b");
```

Comment:
@ces-huynguyen Why you tested two different urls, what is the difference between `/guest/p` and "/guest/b"

Reply:
@tinatian /guest/p throws NPE because of [CPDefinitionAssetDisplayPageFriendlyURLResolver](https://github.com/liferay-core-infra/liferay-portal/blob/master/modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/asset/display/page/portlet/CPDefinitionAssetDisplayPageFriendlyURLResolver.java#L165)
 "/guest/b" throws RTE because of [BlogsLayoutDisplayPageProvider](https://github.com/liferay-core-infra/liferay-portal/blob/master/modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/layout/display/page/BlogsLayoutDisplayPageProvider.java#L103)

Reply:
@ces-huynguyen, for [BlogsLayoutDisplayPageProvider](https://github.com/liferay-core-infra/liferay-portal/blob/master/modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/layout/display/page/BlogsLayoutDisplayPageProvider.java#L103), the root is blog codes transfer PortalException to RuntimeException due to blog api LayoutDisplayPageProvider design. We should fix the issue from blogs side and let it throws PortalException. This is different issue. You may fix it in one new ticket if necessary.